common/pkg/capi/clustertopology/handlers/mutation/meta.go

```diff
@@ -59,6 +68,18 @@ func (mgp metaGeneratePatches) GeneratePatches(
 	resp *runtimehooksv1.GeneratePatchesResponse,
 ) {
 	clusterKey := handlers.ClusterKeyFromReq(req)
+	getCluster := func(ctx context.Context) (*clusterv1.Cluster, error) {
```

Comment:
Can you please provide the motivation I'm not sure I understand.


We already have the client and the clusterKey (from the request), would it not be simpler to have a helper function similar to `ClusterKeyFromReq` that can be used only in the handlers that need it?

Reply:
We don't have the client for all the handlers it's something optional that isn't required. This makes it so that getting the cluster is called once across all the providers and so we don't run into concurrency issues

https://nutanix.slack.com/archives/C069QMDB2QM/p1713193239746189?thread_ts=1712965566.988039&cid=C069QMDB2QM